src/SFA.DAS.ApplyService.Web/Services/ReapplicationCheckService.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using SFA.DAS.ApplyService.Domain.Entities;
+using SFA.DAS.ApplyService.Web.Infrastructure;
+
+namespace SFA.DAS.ApplyService.Web.Services
+{
+    public class ReapplicationCheckService : IReapplicationCheckService
+    {
+        private readonly IApplicationApiClient _applicationApiClient;
+
+
+        public ReapplicationCheckService(IApplicationApiClient applicationApiClient)
+        {
+            _applicationApiClient = applicationApiClient;
+        }
+
+        public async Task<bool> ReapplicationAllowed(Guid signInId, Guid? organisationId)
+        {
+            if (organisationId == null)
+                return true;
+
+            var applications = await _applicationApiClient.GetApplications(signInId, true);
+            var application = applications.OrderByDescending(x => x.UpdatedAt)
```

Comment:
Needs Null Coalesce to stop a potential Null Reference Exception
`applications?.OrderByDescending(x => x.UpdatedAt)`